### PR TITLE
Move CatalogOperationsTest to `:openhouse-spark-itest` module 

### PR DESCRIPTION
**Pull request by HotSushi**
## Summary
Moving this code around for easier navigation and better craftsmanship.

Before this PR there were 2 places for spark itests:
```
openhouse-spark-runtime
openhouse-spark-itest
```
now there's just one
```
openhouse-spark-itest
```

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [X Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

✅ build should succeed
- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: integrations/spark/openhouse-spark-itest/build.gradle [hunk 1]
```diff
@@ -41,10 +43,18 @@ task statementTest(type: Test) {
   }
 }
 
+task catalogTest(type: Test) {
```
Comment (autumnust) [inline integrations/spark/openhouse-spark-itest/build.gradle:46]: why is this needed?

Comment (HotSushi) [reply integrations/spark/openhouse-spark-itest/build.gradle:46]: To order the tests to run after each other. Each of the sparkSessions from statementTest, CatalogTest, SparkBase test conflict with each other.

**Review by autumnust — DISMISSED** [review]
thanks for making this change !

**Review by autumnust — APPROVED** [review]
thanks for pushing extra mileage !
